### DISCUSSION TITLE
Bungee pro and basic version web

### DISCUSSION THREAD

**Discussion by ktech643** [bungee-audio-stretch/bungee#12] [General]
How to get pro version and what is the pricing. And can you guide how to integrate bungee basic in web

**Comment by kupix** [comment]
Bungee Pro: it's a commercial product: please send an email to bungee@parabolaresearch.com to find out what's available and more about pricing.

Bungee Basic in web: so much depends on what you are trying to achieve. For example what are your audio inputs and outputs, do you want only pitch shifting or time stretching too, do you need any other functionality like mixing or sample rate adjustment, etc. Also it depends on the browser/apps/sites you want to support. Typically you need:
- some calling/management C++ code to wrap the Bungee library into, for example, AudioWorklet. This may also handle audio input and output. It would implement a JS API, for example using Emscripten.
- a build system to build to wasm
- JS code to load that wasm
- management code in JS to enable audio loading / input, to talk to the and to implement your UI
- a web UI

We have this as closed source for our web demo and it is available to Bungee Pro customers. It's difficult to give more specific guidance for web integration: there is quite a lot of public information about Emscripten, AudioWorklet, etc. that you can find online.
